PizzaDesignPattren/PizzaDesignPattren/Program.cs

```diff
@@ -0,0 +1,230 @@
+        }
+        public void BuildTopping()
+        {
+            this._thin.PreparePizza("Add favorite topping .. cheese");
+        }
+        public ThinPizza GetPizza()
+        {
+            ThinPizza final = this._thin;
+            this.Reset();
+            return final;
+        }
+    }
+    public class ThickPizzaBuilder : IPizzaBuilder
+    {
+        private ThickPizza _thick = new ThickPizza();
+
+        public ThickPizzaBuilder()
+        {
+            this.Reset();
+        }
+        public void Reset()
+        {
+            this._thick = new ThickPizza();
+        }
+        public void BuildDough()
+        {
+
+            this._thick.PreparePizza("Preparing the dough ..wheat");
+        }
+        public void BuildSauce()
+        {
+            this._thick.PreparePizza("Add souce ..without");
+        }
+        public void BuildTopping()
+        {
+            this._thick.PreparePizza("Add favorite topping ..vegetables + cheese");
+        }
+        public ThickPizza GetPizza()
+        {
+            ThickPizza final = this._thick;
+            this.Reset();
+            return final;
+        }
+    }
+    public class ThinPizza
+    {
+        private List<object> _pizza = new List<object>();
+
+        public void PreparePizza(string pizza)
+        { //determine type of dough ,souce,topping
+            this._pizza.Add(pizza);
+        }
+        public string PizzaOrder()
+        {
+            string str = "";
+            for (int i = 0; i < this._pizza.Count; i++)
+            {
+                str += this._pizza[i] + "\n";
+            }
+            return "Thin Pizza order : \n" + str;
+        }
+    }
+
+    public class ThickPizza
+    {
+        private List<string> _pizza = new List<string>();
+
+        public void PreparePizza(string pizza)
+        { //determine type of dough ,souce,topping
+            this._pizza.Add(pizza);
+        }
+        public string PizzaOrder()
+        {
+            string str = "";
+            for (int i = 0; i < this._pizza.Count; i++)
+            {
+                str += this._pizza[i] + "\n";
+            }
+            return "Thick Pizza order : \n" + str;
+        }
+
+    }
+
+    public class Director
+    {
+        private IPizzaBuilder _pizzaBuilder;
+        public IPizzaBuilder PizzaBuilder
+        {
+            set { this._pizzaBuilder = value; }
+        }
+        public void makePizza()
+        {
+            this._pizzaBuilder.BuildDough();
+            this._pizzaBuilder.BuildSauce();
+            this._pizzaBuilder.BuildTopping();
+        }
+        public void makePizzaWithoutSauce()
+        {
+            this._pizzaBuilder.BuildDough();
+            this._pizzaBuilder.BuildTopping();
+        }
+    }
+
+
+
+
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Director director = new Director();
+            ThickPizzaBuilder thick = new ThickPizzaBuilder();
```

Comment:
It would be better if you used polymorphism to define builders, without creating new builder variable for each one.